processing/src/main/java/io/druid/segment/ByteBuffers.java

```diff
@@ -0,0 +1,147 @@
+package io.druid.segment;
+
+import com.google.common.base.Preconditions;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class ByteBuffers
+{
+  private static final Unsafe UNSAFE;
+  private static final long ADDRESS_OFFSET;
+
+  static {
```

Comment:
I'm not sure we should move this block into the main code until we actually use it outside of the benchmark, it could cause an unwanted exception when running on a vm that doesn't allow unsafe methods.